### PR TITLE
Output of chargemol file developed by Hossam Elgabarty

### PR DESCRIPTION
**Pull request by robertschade**
<hossam.elgabaty@uni-paderborn.de>

Co-authored-by: Hossam Elgabarty <hossam.elgabarty@uni-paderborn.de>

### CODE REVIEW

**Comment by robertschade** [comment]
Dear All,
why is the Formating check always failing for src/qs_chargemol.f even if my local pre-commit hook passes:
image

Is it possible to find out what https://storage.googleapis.com/cp2k-ci/run-cp2k-formatting-a91cc060_report.txt
changed about the file?

**Comment by oschuett** [comment]
Try running `make -j pretty`. Alternatively, you can just apply this diff:
```
diff --git a/src/qs_chargemol.F b/src/qs_chargemol.F
index a3aaba4db..15d312974 100644
--- a/src/qs_chargemol.F
+++ b/src/qs_chargemol.F
@@ -189,9 +189,9 @@ CONTAINS
             END IF
          END IF
 
-       !!---------------------------------------------------------------------------------!
+         !!---------------------------------------------------------------------------------!
          ! Initial parsing of topology and basis set, check maximum l .le. chargemol_lmax
-       !!---------------------------------------------------------------------------------!
+         !!---------------------------------------------------------------------------------!
          num_atoms = SIZE(particle_set)
          ALLOCATE (atoms_cart(3, num_atoms))
          ALLOCATE (atom_symbols(num_atoms))
@@ -510,9 +510,9 @@ CONTAINS
          ! ===========================================================================!
          ! Primitive Exponents
          ! ===========================================================================!
-       !! NOTE: CP2K orders the atomic orbitals as follows:
-       !! Cartesian orbitals: x (slowest loop), y, z (fastest loop)
-       !! Spherical orbitals: m = -l, ..., 0, ..., +l
+         !! NOTE: CP2K orders the atomic orbitals as follows:
+         !! Cartesian orbitals: x (slowest loop), y, z (fastest loop)
+         !! Spherical orbitals: m = -l, ..., 0, ..., +l
 
          WRITE (iwfx, "(A)") "<Primitive Exponents>"
          DO iatom = 1, num_atoms
```

**Comment by robertschade** [comment]
Hi,
running `make -j pretty` locally on the commit e13168050a020a7b728c8fc9a60b282f03af5f89 gives the output:
```
prettified qs_active_space_types.F
1 file(s) prettified, 0 file(s) left unchanged.
0 file(s) prettified, 1 file(s) left unchanged.
prettified libcp2k.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified realspace_grid_cube.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified mp2_gpw.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified mp2_integrals.F
1 file(s) prettified, 0 file(s) left unchanged.
0 file(s) prettified, 1 file(s) left unchanged.
prettified qs_active_space_methods.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified qs_tensors.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified integrator.F
1 file(s) prettified, 0 file(s) left unchanged.
prettified hfx_types.F
1 file(s) prettified, 0 file(s) left unchanged.
```

Even running it explicitly on src/qs_chargemol.F doesn't change anything:
```
cd /scratch/pc2-mitarbeiter/rschade/cp2k_hossam/src/; /scratch/pc2-mitarbeiter/rschade/cp2k_hossam/tools/prettify/prettify.py --do-backup --backup-dir=/scratch/pc2-mitarbeiter/rschade/cp2k_hossam/obj/prettified qs_chargemol.F
0 file(s) prettified, 1 file(s) left unchanged.
```

I am trying it with the patch now.

**Comment by oschuett** [comment]
Hmm, strange. Some time ago `cp2k/tools/prettify/fprettify` was turned into a git submodule. Maybe that's the culprit?

**Comment by robertschade** [comment]
I don't think so, I have recursively cloned a completely new copy of cp2k for this pull request.
Nevertheless, I would like to apologize for the inconvenience caused.

File: src/qs_chargemol.F [hunk 1]
```diff
@@ -0,0 +1,969 @@
...
+      TYPE(particle_type), DIMENSION(:), POINTER         :: particle_set
+      TYPE(qs_energy_type), POINTER                      :: energy
+      TYPE(qs_force_type), DIMENSION(:), POINTER         :: force
+      TYPE(qs_kind_type), DIMENSION(:), POINTER          :: qs_kind_set
+      TYPE(qs_scf_env_type), POINTER                     :: scf_env
+      TYPE(qs_subsys_type), POINTER                      :: subsys
+      TYPE(scf_control_type), POINTER                    :: scf_control
+      TYPE(section_vals_type), POINTER                   :: chargemol_section
+
+!REAL(KIND=dp), PARAMETER                           :: gcc_eps = EPSILON(1.0_dp)
```
Comment (oschuett) [inline src/qs_chargemol.F:116]: Left over?

**Comment by oschuett** [comment]
Maybe you can add a regtest or modify an existing one to execute the new code? Otherwise, LGTM.

**Comment by robertschade** [comment]
I have already considered that. However, I didn't find a possibility to create a suitable test because the code change only adds functionality to export data to a file. In https://github.com/cp2k/cp2k/blob/master/tests/TEST_TYPES I couldn't find the possibility to check stuff from files other than the stdout. Did I miss or miss understand something there? 
I have considered writing something from the new output file into the stdout just to be able to create a regtest but I don't think it is a good idea to do so. What do you think?

**Comment by oschuett** [comment]
> I couldn't find the possibility to check stuff from files other than the stdout.

Yes, that's indeed a shortcoming of our testing framework. Nevertheless, just executing the code already checks it for out-of-bounds, overflows, asserts, etc.

**Comment by juerghutter** [comment]
Is this PR ready for merge? You could update it now or in a follow-up.

**Pull request merged**